cin_validator/rules/cin2022_23/rule_2885.py

```diff
@@ -0,0 +1,360 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import CINTable, RuleContext, rule_definition
+from cin_validator.test_engine import run_rule
+from cin_validator.utils import make_census_period
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+
+ChildProtectionPlans = CINTable.ChildProtectionPlans
+CINdetails = CINTable.CINdetails
+Section47 = CINTable.Section47
+
+CPPstartDate = ChildProtectionPlans.CPPstartDate
+# TODO is it necessary to get LAchildID per table involved?
+LAchildID = ChildProtectionPlans.LAchildID
+CINdetailsID = ChildProtectionPlans.CINdetailsID
+# TODO is it redundant to have both of them here?
+DateOfInitialCPC = Section47.DateOfInitialCPC
+DateOfInitialCPC = CINdetails.DateOfInitialCPC
+
+# Reference date in header is needed to define the period of census.
+Header = CINTable.Header
+ReferenceDate = Header.ReferenceDate
+
+# define characteristics of rule
+@rule_definition(
+    # write the rule code here, in place of 2885
+    code=2885,
+    # replace ChildProtectionPlans with the value in the module column of the excel sheet corresponding to this rule .
+    # Note that even if multiple tables are involved, one table will be named in the module column.
+    module=CINTable.ChildProtectionPlans,
+    # replace the message with the corresponding value for this rule, gotten from the excel sheet.
+    message="Child protection plan shown as starting a different day to the initial child protection conference",
+    # The column names tend to be the words within the < > signs in the github issue description.
+    affected_fields=[
+        CPPstartDate,
+        DateOfInitialCPC,
+    ],  # TODO How can we indicate that the DateOfInitialCPC comes from both tables. Is it necessary?
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # PREPARING DATA
+
+    # Replace ChildProtectionPlans with the name of the table you need.
+    df_cpp = data_container[ChildProtectionPlans].copy()
+    # TODO Is it necessary to copy? How else can we ensure that the original data is not changed in each rule.?.
```

Comment:
It's inside a function so it should change the original data unless you return it, right?

Reply:
That's a fair argument.

Reply:
I think it's safest to use .loc[] in instances like this, right?

Reply:
From what I know, .loc tends to be used with DataFrame-like structures. Is it used with dictionary-like structures too?.
data_container is a dictionary-like mapping so it is flat, no rows and columns.

Reply:
I've converted this to draft so that I don't mistakenly merge it. There are a few changes that have to be made.

Reply:
Oh I'm a fool, you're right, I just assumed it was a df when clearly it isn't